path_data.go

```diff
@@ -169,6 +170,80 @@ func (b *versionedKVBackend) pathDataRead() framework.OperationFunc {
 	}
 }
 
+// validateCheckAndSetOption will validate the cas flag from the options map
+// provided. The cas flag must be provided if required based on the engine's
+// config or the secret's key metadata. If provided, the cas value must match
+// the current version of the secret as denoted by its key metadata entry.
+func validateCheckAndSetOption(data *framework.FieldData, config *Configuration, meta *KeyMetadata) error {
+	var casRaw interface{}
+	var casOk bool
+	optionsRaw, ok := data.GetOk("options")
+	if ok {
+		options := optionsRaw.(map[string]interface{})
+
+		// Verify the CAS parameter is valid.
+		casRaw, casOk = options["cas"]
+	}
+
+	if casOk {
+		var cas int
+		if err := mapstructure.WeakDecode(casRaw, &cas); err != nil {
+			return errors.New("error parsing check-and-set parameter")
+		}
+		if uint64(cas) != meta.CurrentVersion {
+			return errors.New("check-and-set parameter did not match the current version")
+		}
+	} else if config.CasRequired || meta.CasRequired {
+		return errors.New("check-and-set parameter required for this call")
+	}
+
+	return nil
+}
+
+// cleanupOldVersions is responsible for cleaning up old versions. Once a key
+// has more than the configured allowed versions the oldest version will be
+// permanently deleted. A list of version keys to delete will be created.
+// Indices will be ordered such that the oldest version is at the end of the
+// list. Deletes will be performed back-to-front. If there is an error deleting
+// one of the keys, the can remaining keys will be deleted on the next go around.
```

Comment:
```suggestion
// one of the keys, the remaining keys will be deleted on the next go around.
```

Reply:
Ah, thanks for the correction. Will make this manually with other changes outside of GitHub.